cgroup2/manager.go

```diff
@@ -550,78 +550,61 @@ func (c *Manager) MoveTo(destination *Manager) error {
 }
 
 func (c *Manager) Stat() (*stats.Metrics, error) {
-	controllers, err := c.Controllers()
-	if err != nil {
-		return nil, err
-	}
-	// Sizing this avoids an allocation to increase the map at runtime;
-	// currently the default bucket size is 8 and we put 40+ elements
-	// in it so we'd always end up allocating.
-	out := make(map[string]uint64, 50)
-	for _, controller := range controllers {
-		switch controller {
-		case "cpu", "memory":
-			if err := readKVStatsFile(c.path, controller+".stat", out); err != nil {
-				if os.IsNotExist(err) {
-					continue
-				}
-				return nil, err
-			}
-		}
-	}
-	memoryEvents := make(map[string]uint64)
-	if err := readKVStatsFile(c.path, "memory.events", memoryEvents); err != nil {
+	var metrics stats.Metrics
+
+	cpuStat := make(map[string]uint64)
+	if err := readKVStatsFile(c.path, "cpu.stat", cpuStat); err != nil {
 		if !os.IsNotExist(err) {
 			return nil, err
 		}
 	}
-
-	var metrics stats.Metrics
-	metrics.Pids = &stats.PidsStat{
-		Current: getStatFileContentUint64(filepath.Join(c.path, "pids.current")),
-		Limit:   getStatFileContentUint64(filepath.Join(c.path, "pids.max")),
-	}
 	metrics.CPU = &stats.CPUStat{
-		UsageUsec:     out["usage_usec"],
-		UserUsec:      out["user_usec"],
-		SystemUsec:    out["system_usec"],
-		NrPeriods:     out["nr_periods"],
-		NrThrottled:   out["nr_throttled"],
-		ThrottledUsec: out["throttled_usec"],
+		UsageUsec:     cpuStat["usage_usec"],
+		UserUsec:      cpuStat["user_usec"],
+		SystemUsec:    cpuStat["system_usec"],
+		NrPeriods:     cpuStat["nr_periods"],
+		NrThrottled:   cpuStat["nr_throttled"],
+		ThrottledUsec: cpuStat["throttled_usec"],
 		PSI:           getStatPSIFromFile(filepath.Join(c.path, "cpu.pressure")),
 	}
+	memoryStat := make(map[string]uint64, 40)
+	if err := readKVStatsFile(c.path, "memory.stat", memoryStat); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
```

Comment:
So, while looking at this PR (and I think some of this is an existing issue) I wonder if it would make sense to skip assigning values in situations where the file does _not_ exist, or if we should skip all of that if it's not there, and explicitly construct an empty `&stats.MemoryStat{}`.